browser/src/canvas/sections/CommentListSection.ts

```diff
@@ -2060,21 +2062,20 @@ export class CommentSection extends CanvasSectionObject {
 		this.disableLayoutAnimation = false;
 	}
 
-	private layout (immediate: any = null, relayout: boolean = true): void {
-		if (immediate)
-			this.doLayout(relayout);
-		else if (!this.sectionProperties.layoutTimer) {
-			this.sectionProperties.layoutTimer = setTimeout(() => {
-				delete this.sectionProperties.layoutTimer;
-				this.doLayout(relayout);
-			}, 10 /* ms */);
-		} // else - avoid excessive re-layout
+	private update (reLayout: boolean = true): void {
+		this.sectionProperties.pendingUpdate = true;
+		this.sectionProperties.reLayout = reLayout;
+		this.containerObject.addDOMUpdaterCallback(this.updateDOM.bind(this));
```

Comment:
I wonder if each Section would do better to have its own 'updateDOM' method - that checks to see if anything was changed from last time and if not does nothing; otherwise does its updates - and do this through the SectionContainer object hierarcy ?

Then we could leave the queueing of arbitrary callbacks for more legacy JS / DOM pieces? We really want to split the state which we store in JS - from the UI - which we push into the DOM only when it changes and just before render I think ?

Reply:
Yes i tried to split DOM and JS states. I couldn't decide on updateDOM method :) But it's a better way of doing this. I couldn't decide, because many of the sections don't update DOM, so i thought it might be an overkill to call updateDOM function for every inidividual section. Most of them will do nothing. But you are right, let me add it that way.

Reply:
It would be worth watching In The Loop - https://www.youtube.com/watch?v=cCOL7MC4Pl0 to understand what to do. We want to touch the DOM in the animation frame callback.